src/components/anchor-navigation/anchor-navigation.component.js

```diff
@@ -21,35 +24,59 @@ const AnchorNavigation = ({ children, stickyNavigation, styleOverride }) => {
 
   const navigationRef = useRef();
 
-  const setSelectedAnchorBasedOnScroll = useCallback(throttle(() => {
-    const topEdgeOffsets = sectionRefs.current.map(({ current }) => current.getBoundingClientRect().top);
+  const isUserScroll = useRef(true);
+
+  const isUserScrollTimer = useRef();
+
+  const setSelectedAnchorBasedOnScroll = useCallback(() => {
+    const sectionsTopOffsets = sectionRefs.current.map(({ current }) => current.getBoundingClientRect().top);
     const { top: navTopOffset } = navigationRef.current.getBoundingClientRect();
-    const indexOfSmallestNegativeTopOffset = topEdgeOffsets.reduce((currentTopIndex, topOffset, index) => {
-      if (topOffset > navTopOffset) return currentTopIndex;
-      return topOffset > topEdgeOffsets[currentTopIndex] ? index : currentTopIndex;
+
+    const indexOfSmallestNegativeTopOffset = sectionsTopOffsets.reduce((currentTopIndex, sectionTopOffset, index) => {
+      if (sectionTopOffset - SECTION_VISIBILITY_OFFSET > navTopOffset) return currentTopIndex;
+      return sectionTopOffset > sectionsTopOffsets[currentTopIndex] ? index : currentTopIndex;
     }, 0);
 
     setSelectedIndex(indexOfSmallestNegativeTopOffset);
-  }, 200), []);
+  }, []);
```

Comment:
you should decide to either drop useCallback or specify the dependencies: https://reactjs.org/docs/hooks-reference.html#usecallback